src/bsoncxx/array/view.cpp

```diff
@@ -126,7 +126,7 @@ view::const_iterator view::find(std::uint32_t i) const {
     }
 
     if (!bson_iter_init_find(&iter, &b, key.c_str())) {
```

Comment:
Suggest also updating the `return cend();` calls above in this function with `const_iterator(element(key.c_str()))` to include the key in the exception.